### PR TITLE
feat:  Use custom hash.Hash in hmac.HMACStrategy

### PR DESCRIPTION
**Pull request by mash**
Add `Hash func() hash.Hash` field in `hmac.HMACStrategy` to allow using a custom `hash.Hash` function. When nil, the default `sha512.New512_256` is used.

## Related Issue or Design Document

https://github.com/ory/fosite/issues/654

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

**Review by aeneasr — APPROVED** [review]

File: token/hmac/hmacsha_test.go [hunk 1]
```diff
@@ -131,3 +132,17 @@ func TestValidateWithRotatedKeyInvalid(t *testing.T) {
 
 	require.EqualError(t, new(HMACStrategy).Validate(token), "a secret for signing HMAC-SHA512/256 is expected to be defined, but none were")
 }
+
+func TestCustomHMAC(t *testing.T) {
```
Comment (aeneasr) [inline token/hmac/hmacsha_test.go:136]: Could you please also add a success case? :)

**Review by aeneasr — CHANGES_REQUESTED** [review]
Awesome, thank you for your contribution! This looks pretty good and I have some ideas how to improve it further :)

File: token/hmac/hmacsha_test.go (hunk 1 above)
Comment (mash) [reply token/hmac/hmacsha_test.go:136]: Yes, sure.

**Review by aeneasr — APPROVED** [review]
Awesome, thank you! 🎉 Your contribution makes Ory better :)

**Comment by mash** [comment]
Happy to be able to contribute.

**Pull request merged**